src/ucs/arch/x86_64/cpu.c

```diff
@@ -707,6 +736,324 @@ ucs_status_t ucs_arch_get_cache_size(size_t *cache_sizes)
     return cache_count == UCS_CPU_CACHE_LAST ? UCS_OK : UCS_ERR_UNSUPPORTED;
 }
 
+#if (ENABLE_NT_BUFFER_TRANSFER) && (__AVX__)
+
+#define SWITCH_TO_NTSTORE_SZ (1464)
+#define WORD_SZ     2
+#define DWORD_SZ    4
+#define QWORD_SZ    8
+#define YMM_SZ      32
+
+#ifndef __m256i_u
+typedef long long __m256i_u __attribute__((__vector_size__(32), __aligned__(1)));
+#endif
+
+static UCS_F_ALWAYS_INLINE
+size_t ucs_cpu_nt_dst_buffer_transfer(void *dst, const void *src, size_t len,
+                                      int hint, size_t total_len)
```

Comment:
I kept it as 'int' because I am adding (oring) UCS_ARCH_MEMCPY_NT_DEST to the hint based on total_len.

####
void ucs_x86_nt_buffer_transfer(void *dst, const void *src, size_t len,
                                ucs_arch_memcpy_hint_t hint, size_t total_len)
{
    size_t tail_bytes;

    if (ucs_likely(len <= 128)) {
        goto copy_bytes_le_128;
    }

    /* Give precedence to non temporal destination */
    if (ucs_unlikely(total_len > ucs_global_opts.arch.nt_dest_threshold)) {
        hint |= UCS_ARCH_MEMCPY_NT_DEST;
    }

    if (hint & UCS_ARCH_MEMCPY_NT_DEST) {
        tail_bytes = ucs_x86_nt_dst_buffer_transfer(dst, src, len, hint, total_len);
###

Reply:
so let's make it unsiged because it's flags

Reply:
Done.